v2/tools/generator/internal/codegen/pipeline/testdata/TestAddCrossResourceReferences_HandlesMap/person-v20200101.golden

```diff
@@ -0,0 +1,43 @@
+// Code generated by azure-service-operator-codegen. DO NOT EDIT.
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+package v20200101
+
+import (
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+type Person struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              Person_Spec   `json:"spec,omitempty"`
+	Status            Person_STATUS `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+type PersonList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []Person `json:"items"`
+}
+
+type Person_Spec struct {
+	// FullName: As would be used to address mail
+	FullName   string                                  `json:"fullName,omitempty"`
+	References map[string]genruntime.ResourceReference `armReference:"Ids" json:"references,omitempty"`
```

Comment:
This is the important line here -- we detect the collection and swap the type to `genruntime.ResourceReference`